### ISSUE TITLE
[BUG] [csharp-netcore] DateTime format deserialization

### ISSUE THREAD

**Issue by devhl-labs** [OpenAPITools/openapi-generator#7167]
While using bearer tokens, I am setting Configuration.DateTimeFormat, but that format string is not being used while deserializing. I modified the following to use Configuration.DateTimeFormat while deserializing and it does work. Am I doing something wrong? And is there a way to specify different formats for different properties?

```
public CustomJsonCodec(IReadableConfiguration configuration)
{
    _configuration = configuration;
    _serializerSettings.DateFormatString = _configuration.DateTimeFormat;  // added
}
```

**Comment by wing328** [comment]
Looks like a bug. Do you mind submitting a PR with the suggested fix?

**Comment by devhl-labs** [comment]
Yes I can, but if the end user does not set the configuration.DateFormatString, I'm not positive if this will change the outcome or not.  Here are the defaults:
`yyyy'-'MM'-'dd'T'HH':'mm':'ss.FFFFFFFK  //https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonSerializerSettings_DateFormatString.htm`

`yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fffffffzz  //https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings`

**Comment by wing328** [comment]
The format defined by OAS only covers RFC3339. That's why we make it configurable during runtime so that users can use other date time format as well.

**Comment by devhl-labs** [comment]
Okay, but are the two defaults the same?  Notice one is f and one is F and also K versus zz.
